py/torch_tensorrt/dynamo/conversion/impl/shuffle.py

```diff
@@ -0,0 +1,25 @@
+from typing import Optional, Sequence, Union
+
+import numpy as np
+import torch
+from torch.fx.node import Target
+from torch_tensorrt.dynamo.conversion._ConversionContext import ConversionContext
+from torch_tensorrt.dynamo.conversion.converter_utils import SourceIR, get_trt_tensor
+from torch_tensorrt.fx.converters.converter_utils import set_layer_name
+from torch_tensorrt.fx.types import TRTTensor
+
+
+def reshape(
+    ctx: ConversionContext,
+    target: Union[Target, str],
+    source_ir: Optional[SourceIR],
+    name: str,
+    input: Sequence[Union[TRTTensor, torch.Tensor, np.ndarray]],
```

Comment:
Instead of this, you can restrict the input further to be `input: TRTTensor`, then use `@enforce_tensor_types({0: (TRTTensor,)})` on the converter

Reply:
Ok. So, for all converters with single-value input, do you mean we should use `@enforce_tensor_types({0: (TRTTensor,)})`, making the `input` convert to `TRTTensor` automatically?

Reply:
Yes, only if that converter _requires_ a `TRTTensor`, though. There are some converters like conv which can use numpy arrays or torch Tensors to reduce runtime computations by performing operations on the weights at compile time

Reply:
However, for `input` with a sequence, like [this case](https://github.com/pytorch/TensorRT/pull/2343#discussion_r1346509449), we'd better cast to `TRTTensor` manually?

Reply:
Yes, there is currently not support for sequence inputs, which is why that PR has to call `get_trt_tensor` directly for each input

Reply:
BTW, if we put the type conversion part to decorators, in some scenarios, we may want to directly call a function like `impl.shuffle.reshape(...)`. Is there any potential error?

Reply:
I don't understand the use case which could cause the issue exactly here. This decorator is only intended for converters in the `aten_ops_converters.py` file, and not for `impl` implementations. It is the `aten_ops_converters.py` converter's responsibility to ensure that all called converter implementations are valid for the specified tensor types.